### DISCUSSION TITLE
How to contribute

### DISCUSSION THREAD

**Discussion by Assaviv** [CacheControl/json-rules-engine#405] [General]
Having recently explored open-source initiatives, I am particularly drawn to the capabilities of rules engines and am eager to contribute to relevant projects.  This project aligns well with my interests.

How can I contribute to the project?

**Comment by chris-pardy** [comment]
@Assaviv happy to have any contributions. A few things to note:

* Myself and @CacheControl who are the maintainers on this repo both have full-time jobs and other obligations that don't allow us to put a consistent amount of work into this project. That being said:
  * Any critical security updates will get addressed reasonably quickly.
  * If you have a feature request or a PR to add a feature that seems reasonable we'll try to address it fairly quickly.
  * If feature requests or PRs aren't getting prioritized it's not that we don't value your contribution we're just us prioritizing that work against things outside this project.
* Your best bet in terms of contributing is to find feature requests or make your own and then create a PR for them. If the feature is important to you, well implemented, and doesn't break backwards compatibility or add significant complexity to the system then we'll get it added eventually.
   * One feature that would be nice to have is the ability for a rule to emit a different event/events on success or on failure.
 
 There's a decent amount of technical debt and hard to follow logic in different methods that relate to the gradual evolution of this library. A thorough acceptance testing suite that clearly documents every feature of this library would be a great place to start in terms of building a ground-up rewrite / re-architecture for the system. That's been my on-going project when I have time to spend on it which is not often.
